eos/form-factors/analytic-b-to-gamma-qcdf.cc

```diff
@@ -0,0 +1,867 @@
+            (-1.4190476190476191 + (716 * xsg) / 35. + 2 * eulergamma - 24 * xsg * eulergamma + 2 * eulergamma2 - 12 * xsg * eulergamma2 - (569 * expint_ei_xom_xsgplus) / 105. + (214 * xsg * expint_ei_xom_xsgplus) / 5. - 2 * eulergamma * expint_ei_xom_xsgplus + 12 * xsg * eulergamma * expint_ei_xom_xsgplus - 4 * xom * hyp_xom_xsgplus + 20 * xom * xsg * hyp_xom_xsgplus + 2 * L - 24 * xsg * L + 2 * expint_ei_xom_xsgplus * L - 12 * xsg * expint_ei_xom_xsgplus * L - (1147 * xsg2) / 21. + 70 * eulergamma * xsg2 + 30 * eulergamma2 * xsg2 - 111 * expint_ei_xom_xsgplus * xsg2 - 30 * eulergamma * expint_ei_xom_xsgplus * xsg2 - 36 * xom * hyp_xom_xsgplus * xsg2 + 70 * L * xsg2 + 30 * expint_ei_xom_xsgplus * L * xsg2 + (4160 * xsg3) / 63. - (320 * eulergamma * xsg3) / 3. - 40 * eulergamma2 * xsg3 + (460 * expint_ei_xom_xsgplus * xsg3) / 3. + 40 * eulergamma * expint_ei_xom_xsgplus * xsg3 + 20 * xom * hyp_xom_xsgplus * xsg3 - (320 * L * xsg3) / 3. - 40 * expint_ei_xom_xsgplus * L * xsg3 - (643 * xsg4) / 21. + 86 * eulergamma * xsg4 + 30 * eulergamma2 * xsg4 - 111 * expint_ei_xom_xsgplus * xsg4 - 30 * eulergamma * expint_ei_xom_xsgplus * xsg4 + 20 * xom * hyp_xom_xsgplus * xsg4 + 86 * L * xsg4 + 30 * expint_ei_xom_xsgplus * L * xsg4 + (44 * xsg5) / 35. - (184 * eulergamma * xsg5) / 5. - 12 * eulergamma2 * xsg5 + (214 * expint_ei_xom_xsgplus * xsg5) / 5. + 12 * eulergamma * expint_ei_xom_xsgplus * xsg5 - 36 * xom * hyp_xom_xsgplus * xsg5 - (184 * L * xsg5) / 5. - 12 * expint_ei_xom_xsgplus * L * xsg5 - (log_xsgplus * (-569 + 4494 * xsg - 11655 * xsg2 + 16100 * xsg3 - 11655 * xsg4 + 4494 * xsg5 + 210 * eulergamma * (-1 + 6 * xsg - 15 * xsg2 + 20 * xsg3 - 15 * xsg4 + 6 * xsg5 - xsg6) - 210 * L * (-1 + 6 * xsg 
```

Comment:
@pluegh this is the issue. ``a_begin`` and ``a_end`` used to be **references**. Switching to the capture ``[&]`` (i.e., implicit capture by reference) means that the compiler is supposed to reference all symbols that are used within the lambda and not otherwise specified in the capture. However, references to references can be problematic. The standard committee has adopted a [DR](https://cplusplus.github.io/CWG/issues/2313.html) for this problem.
There is not obvious solution, not even when copying the reference (since its underlying object can already be destroyed).

For the time being, please use the following code instead:
```
auto c_d_star_integrand = [&](const double & x, const auto & a_begin, const auto & a_end) -> double {
[...]
}
[...]
result += integrate<GSL::QAGS>(std::bind(c_d_star_integrand, std::placeholders::_1, a_begin, a_end), 0.0, 1.0);
```
(This is untested)

```

Reply:
Now I see this comment -- that's exactly it. The naive version should work in C++17 afaik, but the above workaround addresses the actual issue better. Thanks!